### PR TITLE
Move to maven and GitHub actions

### PR DESCRIPTION
**Pull request by MancunianSam**
I've added in the standard github workflows and changed build.sbt to push to maven.

I've also changed the presigned url tests slightly.
Because the test is calling the real presign method, it needed
credentials in the environment to work. This wasn't a problem on Jenkins
because they were there but now they won't be on GitHub actions.

Iv'e added in an apply method to pass in a presigner if we want to and
then set dummy credentials in the tests.

This should now run without AWS env vars being set.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 addSbtPlugin("com.github.gseitz" % "sbt-release" % "1.0.13")
 resolvers += Resolver.jcenterRepo
-addSbtPlugin("ohnosequences" % "sbt-s3-resolver" % "0.19.0")
\ No newline at end of file
+addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "3.9.12")
+addSbtPlugin("com.github.sbt" % "sbt-pgp" % "2.1.2")
```
Comment (TomJKing) [inline project/plugins.sbt]: Missing line break

File: src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala [hunk 2]
```diff
@@ -13,6 +12,10 @@ import software.amazon.awssdk.services.s3.S3AsyncClient
 import software.amazon.awssdk.services.s3.model.{GetObjectRequest, GetObjectResponse, PutObjectRequest, PutObjectResponse}
 import uk.gov.nationalarchives.aws.utils.TestUtils.failedFuture
 import cats.effect.unsafe.implicits.global
+import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.s3.presigner.S3Presigner
+import software.amazon.awssdk.services.s3.presigner.model.{GetObjectPresignRequest, PresignedGetObjectRequest}
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala]: Don't think these imports are needed: `import software.amazon.awssdk.services.s3.presigner.model.{GetObjectPresignRequest, PresignedGetObjectRequest}`

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of very minor comments

File: project/plugins.sbt (hunk 1 above)
Comment (MancunianSam) [reply project/plugins.sbt]: Ah yeah sorry, I've added it in.

File: src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala (hunk 2 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/aws/utils/S3UtilsTest.scala]: I've deleted them.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**